File: tp1/utils/distances.py
from typing import Tuple

import numpy as np

def euclidean2D_distance(u: Tuple, v: Tuple) -> int:
    dx = u[0] - v[0]
    dy = u[1] - v[1]

    rounded_distance = np.round(np.sqrt(dx**2 + dy**2))
    return int(rounded_distance)

def pseudo_eclidean2D_distance(u: Tuple, v: Tuple) -> int:
    dx = u[0] - v[0]
    dy = u[1] - v[1]

    real_distance = np.sqrt((dx**2 + dy**2) / 10)
    rounded_distance = np.round(real_distance)

    if rounded_distance < real_distance:
        return rounded_distance + 1

    return int(rounded_distance)

File: tp3/src/distances.c
#include <math.h>
#include "distances.h"

int euclid_2d(point_t u, point_t v) {
    double dx = u.x - v.x;
    double dy = u.y - v.y;

    return round(sqrt(dx*dx + dy*dy));
}

int pseudo_euclid_2d(point_t u, point_t v) {
    double dx = u.x - v.x;
    double dy = u.y - v.y;

    double real_distance = sqrt((dx*dx + dy*dy) / 10.0);
    double rounded_distance = round(real_distance);

    if (rounded_distance < real_distance)
        rounded_distance++;

    return rounded_distance;
}